controllers/internal/utils/annotations.go

```diff
@@ -0,0 +1,17 @@
+package utils
+
+// SaveAnnotations adds or updates annotations from a2 from a1 and returns merge
+func SaveAnnotations(a1 map[string]string, a2 map[string]string) map[string]string {
```

Comment:
consider renaming to emphasize merge priority:
```suggestion
func MergeAnnotations(target map[string]string, source map[string]string) map[string]string {
```
or
```suggestion
func MergeAnnotations(to map[string]string, from map[string]string) map[string]string {
```